Sources/StreamChat/Controllers/PollController/PollController.swift

```diff
@@ -0,0 +1,283 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import CoreData
+import Foundation
+
+public extension ChatClient {
+    /// Creates and returns a `PollController` for the specified message and poll.
+    ///
+    /// - Parameters:
+    ///   - messageId: The unique identifier of the message associated with the poll.
+    ///   - pollId: The unique identifier of the poll.
+    /// - Returns: A `PollController` initialized with the provided client, message ID, and poll ID.
+    func pollController(messageId: MessageId, pollId: String) -> PollController {
+        .init(client: self, messageId: messageId, pollId: pollId)
+    }
+}
+
+public class PollController: DataController, DelegateCallable, DataStoreProvider {
+    /// The `ChatClient` instance this controller belongs to.
+    public let client: ChatClient
+
+    /// The identifier of the message the poll belongs to.
+    public let messageId: MessageId
+
+    /// The identifier of the poll this controllers represents.
+    public let pollId: String
+    
+    public var poll: Poll? {
+        startObserversIfNeeded()
+        return pollObserver?.item
+    }
+    
+    public var ownVotes: LazyCachedMapCollection<PollVote> {
+        ownVotesObserver.items
+    }
+    
+    private let pollsRepository: PollsRepository
+    
+    /// Set the delegate of `ChannelController` to observe the changes in the system.
+    public var delegate: PollControllerDelegate? {
+        get { multicastDelegate.mainDelegate }
+        set { multicastDelegate.set(mainDelegate: newValue) }
+    }
+
+    /// A type-erased delegate.
+    internal var multicastDelegate: MulticastDelegate<PollControllerDelegate> = .init() {
+        didSet {
+            stateMulticastDelegate.set(mainDelegate: multicastDelegate.mainDelegate)
+            stateMulticastDelegate.set(additionalDelegates: multicastDelegate.additionalDelegates)
+        }
+    }
+    
+    let ownVotesQuery: PollVoteListQuery
+    
+    private var pollObserver: EntityDatabaseObserverWrapper<Poll, PollDTO>?
```

Comment:
Is there any specific reason why both observers are not created with `lazy var` properties?

Reply:
Nope, I will make them lazy